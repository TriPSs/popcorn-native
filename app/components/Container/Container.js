import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, View } from 'react-native'

import colors from 'modules/colors'

export const styles = StyleSheet.create({

  root: {
    position: 'relative',
    backgroundColor: colors.BACKGROUND,
    zIndex: 2,
  },

  elevationRoot: {
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    backgroundColor: '#ffffff',
    zIndex: 1,
  },

  elevation0: {
    opacity: 0,
  },

  elevation1: {
    opacity: 0.05,
  },

  elevation2: {
    opacity: 0.07,
  },

  elevation3: {
    opacity: 0.08,
  },

  elevation4: {
    opacity: 0.09,
  },

  elevation6: {},

  elevation8: {},

  elevation12: {},

  elevation16: {},

  elevation24: {},
})

export const Container = ({ children, elevation, style }) => (
  <View
    style={[styles.root, style]}
    elevation={elevation}
    pointerEvents={'box-none'}>
    {elevation > 0 && (
      <View
        style={[
          styles.elevationRoot,
          styles[`elevation${elevation}`],
        ]}
        pointerEvents={'none'} />
    )}

    {children}
  </View>
)

Container.propTypes = {
  children: PropTypes.node,
  elevation: PropTypes.number,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ]),
}

Container.defaultProps = {
  children: null,
  elevation: 0,
  component: View,
  style: null,
}

export default Container
