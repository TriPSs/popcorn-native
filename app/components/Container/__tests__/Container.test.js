import React from 'react'
import { shallow } from 'enzyme'

import Container from '../Container'

describe('(Component) Container', () => {

  it('should render with defaults', () => {
    const result = shallow(
      <Container>
        test
      </Container>,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with elevation 1', () => {
    const result = shallow(
      <Container
        elevation={1}>
        test
      </Container>,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with elevation 2 and custom style', () => {
    const result = shallow(
      <Container
        style={{ margin: 20 }}
        elevation={1}>
        test
      </Container>,
    )

    expect(result).toMatchSnapshot()
  })

})
