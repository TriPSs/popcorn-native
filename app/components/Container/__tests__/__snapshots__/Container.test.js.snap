// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Container should render with defaults 1`] = `
<View
  elevation={0}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "backgroundColor": "#202125",
        "position": "relative",
        "zIndex": 2,
      },
      null,
    ]
  }
>
  test
</View>
`;

exports[`(Component) Container should render with elevation 1 1`] = `
<View
  elevation={1}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "backgroundColor": "#202125",
        "position": "relative",
        "zIndex": 2,
      },
      null,
    ]
  }
>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        },
        Object {
          "opacity": 0.05,
        },
      ]
    }
  />
  test
</View>
`;

exports[`(Component) Container should render with elevation 2 and custom style 1`] = `
<View
  elevation={1}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "backgroundColor": "#202125",
        "position": "relative",
        "zIndex": 2,
      },
      Object {
        "margin": 20,
      },
    ]
  }
>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        },
        Object {
          "opacity": 0.05,
        },
      ]
    }
  />
  test
</View>
`;
